kubernetes/cache/cache.go

```diff
@@ -382,6 +385,20 @@ func (c *kialiCacheImpl) SetNamespace(token string, namespace models.Namespace)
 	c.namespaceStore.Set(key, ns)
 }
 
+// DeleteNamespace should be called when it has been detected that a namespace was deleted from a particular cluster.
+// This cleans up some internals related to the namespace that is now gone.
+func (c *kialiCacheImpl) DeleteNamespace(cluster string, namespace string, token string) {
+	c.namespacesLock.Lock()
+	defer c.namespacesLock.Unlock()
+
+	key := namespacesKey{cluster: cluster, token: token}
+	c.namespaceStore.Remove(key)
+
+	if kubeCache, err := c.GetKubeCache(cluster); err == nil {
+		kubeCache.StopNamespace(namespace)
+	}
+}
+
```

Comment:
Might as well delete all the tokens for that namespace/cluster right? Something like `RefreshTokenNamespace(cluster, namespace)`.

Reply:
OK. I put that in the implementation along with the new ErrorHandler stuff. See https://github.com/kiali/kiali/commit/3ce366677cc10a4bd35882f4f6973dab5b979ff3#diff-cfd09f004cc6d39d422ad55dba06932081887b2323b3854596e06a27a6563d74R393